player/tanks.py

```diff
@@ -0,0 +1,112 @@
+    elif my_vehicle.vehicle_type == VehicleType.HEAVY_TANK:
+        return Heavy.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.LIGHT_TANK:
+        return Light.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.SPG:
+        return SPG.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.AT_SPG:
+        return AT_SPG.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    return False
+
+
+class Medium:
+    @staticmethod
+    def is_enemy_in_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle):
+        distance = map_distance(my_vehicle.position, enemy_vehicle.position)
+        if distance == 2:
+             return True
+        return False
+    
+    @staticmethod
+    def move(path: list) -> Hex:
+        moveHex = Hex(0, 0, 0)
+        if len(path) >= 3:
+            moveHex = Hex(*path[2])
+        elif len(path) >= 2:
+            moveHex = Hex(*path[1])
+        return moveHex
+
+
+class Heavy:
+    @staticmethod
+    def is_enemy_in_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle):
+        distance = map_distance(my_vehicle.position, enemy_vehicle.position)
+        if distance <= 2:
+            return True
+        return False
+    
+    @staticmethod
+    def move(path: list) -> Hex:
+        moveHex = Hex(0, 0, 0)
+        if len(path) >= 2:
+            moveHex = Hex(*path[1])
+        return moveHex
+
+
+class Light:
+    @staticmethod
+    def is_enemy_in_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle):
+        distance = map_distance(my_vehicle.position, enemy_vehicle.position)
+        if distance == 2:
+            return True
+        return False
+    
+    @staticmethod
+    def move(path: list) -> Hex:
+        moveHex = Hex(0, 0, 0)
+        if len(path) >= 4:
+            moveHex = Hex(*path[3])
+        elif len(path) >= 3:
+            moveHex = Hex(*path[2])
+        elif len(path) >= 2:
+            moveHex = Hex(*path[1])
+        return moveHex
+    
+
+class SPG:
+    @staticmethod
+    def is_enemy_in_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle):
+        distance = map_distance(my_vehicle.position, enemy_vehicle.position)
+        if distance == 3:
+            return True
+        return False
+    
+    @staticmethod
+    def move(path: list) -> Hex:
+        moveHex = Hex(0, 0, 0)
+        if len(path) >= 2:
+            moveHex = Hex(*path[1])
+        return moveHex
+    
+
+class AT_SPG:
+    @staticmethod
+    def __two_out_of_three(my_position: Hex, enemy_position: Hex):
+        return (my_position.x == enemy_position.x and my_position.y != enemy_position.y and my_position.z != enemy_position.z) or\
```

Comment:
I think we could just add method `hex.on_line_with(other_hex)` which would return `x == other.x or y == other.y or z == other.z`

Reply:
True. It will be changed.